river/Output.zig

```diff
@@ -185,227 +163,259 @@ pub fn sendViewTags(self: Self) void {
-                };
-            }
-        } else {
-            if (position == MasterPosition.Top) { // Top master
-                const slave_width = @divTrunc(layout_width, slave_count);
-                const slave_width_rem = layout_width % slave_count;
-                new_box = .{
-                    .x = @intCast(i32, (i - master_count) * slave_width + if (i > master_count) slave_width_rem else 0),
-                    .y = @intCast(i32, master_stack_size),
-                    .width = slave_width + if (i == master_count) slave_width_rem else 0,
-                    .height = slave_stack_size,
-                };
-            } else if (position == MasterPosition.Right) { // Right master
-                const slave_height = @divTrunc(layout_height, slave_count);
-                const slave_height_rem = layout_height % slave_count;
-                new_box = .{
-                    .x = 0,
-                    .y = @intCast(i32, (i - master_count) * slave_height + if (i > master_count) slave_height_rem else 0),
-                    .width = slave_stack_size,
-                    .height = slave_height + if (i == master_count) slave_height_rem else 0,
-                };
-            } else if (position == MasterPosition.Bottom) { // Bottom master
-                const slave_width = @divTrunc(layout_width, slave_count);
-                const slave_width_rem = layout_width % slave_count;
-                new_box = .{
-                    .x = @intCast(i32, (i - master_count) * slave_width + if (i > master_count) slave_width_rem else 0),
-                    .y = 0,
-                    .width = slave_width + if (i == master_count) slave_width_rem else 0,
-                    .height = slave_stack_size,
-                };
-            } else { // Left master
-                const slave_height = @divTrunc(layout_height, slave_count);
-                const slave_height_rem = layout_height % slave_count;
-                new_box = .{
-                    .x = @intCast(i32, master_stack_size),
-                    .y = @intCast(i32, (i - master_count) * slave_height + if (i > master_count) slave_height_rem else 0),
-                    .width = slave_stack_size,
-                    .height = slave_height + if (i == master_count) slave_height_rem else 0,
+/// Parse a window configuration string and write values to the box
+fn parseWindowConfig(buffer: []const u8, box: *Box) bool {
+    Log.Debug.log("Trying to parse window configuration string \"{}\"", .{buffer});
+    var i: u32 = 0;
+    var it = std.mem.split(buffer, " ");
+    while (it.next()) |token| {
+        Log.Debug.log("Current token: '{}'", .{token});
+        switch (i) {
+            // X
+            0 => {
+                box.x = std.fmt.parseInt(i32, token, 10) catch |err| {
+                    Log.Error.log("Invalid layout: Invalid window X position.", .{});
+                    return false;
                 };
```

Comment:
That function signature won't work. Before calling `parseWindowConfig()`, a new member is added to the array list. We only have the pointer to this member, so we need to work on that. Returning `Box` will make things more complicated.

I'll see what I can do about returning a `LayoutError` enum thingy.

Reply:
This would work fine:
```zig
try view_boxen.append(try parseWindowConfig(buffer));
```

Also realized that I typoed the declaration of `LayoutError` I gave, should be 
```zig
const LayoutError = error{
    ...
};
```

Reply:
If you want to stick with `addOne()` to allow handling the OOM first, we could do
```zig
const neu_box = try view_boxen.addOne();
neu_box.* = try parseWindowConfig(buffer);
```
Don't worry about what looks like an unnecessary copy, Zig's copy elision and/or result location semantics should eliminate it.